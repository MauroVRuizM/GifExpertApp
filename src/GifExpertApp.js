import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Saitama']);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ol>
                {
                    categories.map(data => (
                    <GifGrid
                    key={ data }
                    category={ data } 
                    />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;