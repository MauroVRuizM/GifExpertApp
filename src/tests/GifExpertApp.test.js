import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp/>', () => {
    const wrapper = shallow( < GifExpertApp /> );

    test('Debe mostrarse <GifExpertApp/> correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

})