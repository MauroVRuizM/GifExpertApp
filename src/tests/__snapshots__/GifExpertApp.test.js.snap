// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> Debe mostrarse <GifExpertApp/> correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Saitama"
      key="Saitama"
    />
  </ol>
</Fragment>
`;
